src/renderer.tsx

```diff
@@ -44,4 +59,132 @@ export default function (context): void {
 		statuses.push(BackupStates.creatingDatabaseSnapshot);
 		return statuses;
 	});
+
+	// add new routes and components to Local core
+	hooks.addFilter('AddSiteIndexJS:RoutesArray', (routes, path) => {
+		routes.forEach((route) => {
+			if (route.path === `${path}/`) {
+				route.path = LOCAL_ROUTES.ADD_SITE_CREATE_NEW;
+			}
+		});
+
+		routes.push(
+			{ key: 'add-site-choose', path: `${path}/`, component: ChooseCreateSiteHOC },
+			{ key: 'add-site-select-site-backup', path: LOCAL_ROUTES.ADD_SITE_BACKUP_SITE, component: SelectSiteBackupHOC },
+			{ key: 'add-site-select-snapshot', path: LOCAL_ROUTES.ADD_SITE_BACKUP_SNAPSHOT, component: SelectSnapshotHOC },
+		);
+
+		return routes;
+	});
+
+	// optionally modify NewSiteEnvironment component functionality in Local core
+	hooks.addFilter('AddSiteIndexJS:NewSiteEnvironment', (newSiteEnvironmentProps) => {
+		if (newSiteEnvironmentProps.siteSettings.cloudBackupMeta?.createdFromCloudBackup) {
+			const continueCreateSite = () => {
+				LocalRenderer.sendIPCEvent('addSite', {
+					newSiteInfo: newSiteEnvironmentProps.siteSettings,
+					goToSite: true,
+					installWP: false,
+				});
+			};
+
+			const onGoBack = () => {
+				LocalRenderer.sendIPCEvent('goToRoute', LOCAL_ROUTES.ADD_SITE_BACKUP_SNAPSHOT);
+			};
+
+			newSiteEnvironmentProps.onContinue = continueCreateSite;
+			newSiteEnvironmentProps.onGoBack = onGoBack;
+			newSiteEnvironmentProps.buttonText = 'Restore Site';
+
+			return newSiteEnvironmentProps;
```

Comment:
Instead of mutating the `newSiteEnvironmentProps` object, what do you think about returning something like:

```
return {
       ...newSiteEnvironmentProps,
       onGoBack,
       onContinue: continueCreateSite,
       buttonText: 'Restore Site',
    };
```